source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp

```diff
@@ -1894,5 +1944,14 @@ namespace RendererQml
 
         return function.str();
     }
+
+	const std::string AdaptiveCardQmlRenderer::getActionShowCardClickFunc(std::string buttonId)
+	{
+		std::ostringstream function;
+		function << buttonId << "_loader.visible = !" << buttonId << "_loader.visible";
+		function << "\n" << buttonId << "_dd.source =" <<  buttonId << "_dd.source == "<< "\"" << RendererQml::arrow_up_12 << "\" ?" <<"\"" << RendererQml::arrow_down_12 << "\"" << ":" << "\"" << RendererQml::arrow_up_12 << "\"";
```

Comment:
Since signals like pressed and clicked were not behaving the way we wanted, so the logic for change of icon is onClick function,
Please go through this and let me know if you want it to be done in some other manner